sklearn/ensemble/bagging.py

```diff
@@ -454,14 +483,14 @@ def _validate_estimator(self):
         super(BaggingClassifier, self)._validate_estimator(
             default=DecisionTreeClassifier())
 
-    def _set_oob_score(self, X, y):
+    def _set_oob_score(self, X, y, n_more_estimators):
         n_classes_ = self.n_classes_
         classes_ = self.classes_
         n_samples = y.shape[0]
 
         predictions = np.zeros((n_samples, n_classes_))
 
-        for estimator, samples, features in zip(self.estimators_,
+        for estimator, samples, features in zip(self.estimators_[-n_more_estimators:],
                                                 self.estimators_samples_,
                                                 self.estimators_features_):
```

Comment:
What should the `oob_score_` and `oob_decision_function_` attributes be set to after fitting the second batch in order to signal "this attribute isn't available".

Normally the attributes simply don't exist if `oob_score=False`. Should we set it to `None`, leave it set to the value from the first batch, apply some python black magic to remove the attribute again? I dislike the black magic idea, leaving it set to the value form the first batch will potentially confuse people, so `None`?